frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryPlaceVisitStorage.java

```diff
@@ -60,41 +59,40 @@ public void removePlaceVisit(String tripId, String placeId) throws PlaceVisitNot
 
   @Override
   public Optional<PlaceVisitModel> getPlaceVisit(String tripId, String placeId) {
-    return Optional.ofNullable(storage.get(tripId)).map(placesMap -> placesMap.get(placeId));
+    return Optional.ofNullable(placesByTripIdByPlaceId.get(tripId))
+        .map(placesMap -> placesMap.get(placeId));
   }
 
   @Override
   public boolean updateUserMarkOrAddPlaceVisit(
       PlaceVisitModel placeVisit, PlaceVisitModel.UserMark newStatus) {
-    synchronized (storage) {
-      Map<String, PlaceVisitModel> placesMap = storage.get(placeVisit.tripId());
-      if (placesMap == null) {
-        Map<String, PlaceVisitModel> newPlaceMap = new ConcurrentHashMap<>();
-        newPlaceMap.put(placeVisit.placeId(), placeVisit);
-        return false;
-      }
-      if (placesMap.get(placeVisit.placeId()) != null) {
-        placesMap.put(placeVisit.placeId(),
-            placesMap.get(placeVisit.placeId()).toBuilder().setUserMark(newStatus).build());
-        return true;
+    AtomicBoolean alreadyInStorage = new AtomicBoolean(false);
```

Comment:
What's the purpose in using AtomicBoolean instead of a regular boolean

Reply:
you can't change the value of a variable within a lambda function so I couldn't use a regular boolean variable. The value within the AtomicBoolean can be updated though